__tests__/auth.test.ts

```diff
@@ -1,30 +1,288 @@
-import request from 'supertest'
-import { expect, describe, test } from 'vitest'
-import { gqlApiUrl } from './testSetup.test.js'
-import { searchSubmissionsQuery } from './submissions.test.js'
-import { Submission, SubmissionSearchFilters } from '../src/typeDefs/gqlTypes.js'
-import { gqlRequest } from '../utils/gqlTool.js'
-
-describe('auth', () => {
-    test('can login and access secure routes', async () => {
-        // TODO add auth0 test here
-
-        // test a secure endpoint as a logged in user
-        const searchSubmissionsRequest = {
-            query: searchSubmissionsQuery,
-            variables: {
-                filters: {} satisfies SubmissionSearchFilters
-            }
-        } as gqlRequest
-
-        //Get the submissions query data
-        const searchResult = await request(gqlApiUrl).post('').send(searchSubmissionsRequest)
-
-        //should not have errors
-        expect(searchResult.body.errors).toBeUndefined()
-
-        const searchedSubmissions = searchResult.body.data.submissions as Submission[]
-        
-        expect(searchedSubmissions).toBeDefined()
+import { describe, it, expect, beforeEach, vi } from 'vitest'
+import { authorize, buildUserContext, Role, Scope } from '../src/auth'
+import { envVariables } from '../utils/environmentVariables'
+import { FastifyRequest } from 'fastify'
+
+// Mock envVariables to control testing environment
+beforeEach(() => {
+    vi.restoreAllMocks()
+})
+
+describe('authorize()', () => {
+    it('should skip auth checks in testing environment', () => {
+        vi.spyOn(envVariables, 'isTestingEnvironment').mockReturnValue(true)
+        const result = authorize({ sub: 'test1', name: 'Test', email: 'findadoctest@test.com', roles: [], scope: '' }, [Scope['read:facilities']])
+
+        expect(result).toBe(true)
+    })
+
+    it('should allow Admin to access all admin scopes', () => {
+        vi.spyOn(envVariables, 'isTestingEnvironment').mockReturnValue(false)
+        const adminUser = {
+            sub: 'adminTest',
+            name: 'Findadoc Admin',
+            email: 'findadocAdmin@admin.com',
+            roles: [Role.Admin],
+            scope: ''
+        }
+
+        // Admins should have access to all scopes
+        expect(authorize(adminUser, [Scope['read:healthcareprofessionals']])).toBe(true)
```

Comment:
since scopes will change, for these tests, let's just test one scope that works, and one that shouldn't in the corresponding tests. 
For admin, `write:logs` is admin only right now, so that's a good one to test it should have it, but moderator's cannot.